### ISSUE TITLE
Zod-OpenAPI Middleware wrong path

### ISSUE THREAD

**Issue by PassiDel** [honojs/middleware#129]
Hey! I followed the [docs](https://github.com/honojs/middleware/blob/main/packages/zod-openapi/README.md) for the new OpenAPI middleware, but now I have a problem with the dynamic route parameter. With `/:id` hono works as expected, but the generated OpenAPI JSON is incorrect, since it expects this format: `/{id}`.

If I change the path in `createRoute` to `/{id}` the JSON is valid but hono won't resolve this route.

Is there any way, how I could solve that?

<details>
  <summary>Code</summary>
  
  ```ts
import { serveStatic } from 'hono/bun';
import { createRoute, OpenAPIHono, z } from '@hono/zod-openapi';
import { cors } from 'hono/cors';

const port = parseInt(process.env.PORT) || 3000;

const ParamsSchema = z.object({
  id: z
    .string()
    .min(3)
    .openapi({
      param: {
        name: 'id',
        in: 'path'
      },
      example: '1212121'
    })
});

const UserSchema = z
  .object({
    id: z.string().openapi({
      example: '123'
    }),
    name: z.string().openapi({
      example: 'John Doe'
    }),
    age: z.number().openapi({
      example: 42
    })
  })
  .openapi('User');

const ErrorSchema = z.object({
  code: z.number().openapi({
    example: 400
  }),
  message: z.string().openapi({
    example: 'Bad Request'
  })
});

const route = createRoute({
  method: 'get',
  path: '/users/:id',
  request: {
    params: ParamsSchema
  },
  responses: {
    200: {
      content: {
        'application/json': {
          schema: UserSchema
        }
      },
      description: 'Retrieve the user'
    },
    400: {
      content: {
        'application/json': {
          schema: ErrorSchema
        }
      },
      description: 'Returns an error'
    }
  }
});

const app = new OpenAPIHono();

app.use('*', cors());

app.use('/favicon.ico', serveStatic({ path: './public/favicon.ico' }));

app.get('/', (c) => {
  return c.json({ message: 'Hello World!' });
});

app.doc('/doc', {
  openapi: '3.0.0',
  info: {
    version: '1.0.0',
    title: 'My API'
  },
  servers: [
    {
      url: 'http://localhost:3000',
      description: 'Local dev'
    }
  ]
});

const routes = app.openapi(route, (c) => {
  const { id } = c.req.valid('param');
  return c.jsonT({
    id,
    age: 20,
    name: 'Ultra-man'
  });
});

console.log(`Running at http://localhost:${port}`);

export default {
  port,
  fetch: app.fetch
};

```

</details>

<details>
	<summary>OpenAPI JSON</summary>

```json
{
  "openapi": "3.0.0",
  "info": {
    "version": "1.0.0",
    "title": "My API"
  },
  "servers": [
    {
      "url": "http://localhost:3000",
      "description": "Local dev"
    }
  ],
  "components": {
    "schemas": {
      "User": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string",
            "example": "123"
          },
          "name": {
            "type": "string",
            "example": "John Doe"
          },
          "age": {
            "type": "number",
            "example": 42
          }
        },
        "required": [
          "id",
          "name",
          "age"
        ]
      }
    },
    "parameters": {}
  },
  "paths": {
    "/users/:id": {
      "get": {
        "parameters": [
          {
            "schema": {
              "type": "string",
              "minLength": 3,
              "example": "1212121"
            },
            "required": true,
            "name": "id",
            "in": "path"
          }
        ],
        "responses": {
          "200": {
            "description": "Retrieve the user",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/User"
                }
              }
            }
          },
          "400": {
            "description": "Returns an error",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "code": {
                      "type": "number",
                      "example": 400
                    },
                    "message": {
                      "type": "string",
                      "example": "Bad Request"
                    }
                  },
                  "required": [
                    "code",
                    "message"
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
}
```
</details>

**Comment by yusukebe** [comment]
Hi @PassiDel !

Thanks for raising the issue. It's may be fixed in #132. I'll release the new version which includes this fix.

**Comment by PassiDel** [comment]
Damn, that was fast! Thank you very much
